membrane_curvature/base.py

```diff
@@ -0,0 +1,68 @@
+import numpy as np
+
+from membrane_curvature.surface import get_z_surface
+from membrane_curvature.curvature import mean_curvature, gaussian_curvature
+
+
+from MDAnalysis.analysis.base import AnalysisBase
+import MDAnalysis as mda
+
+
+class MembraneCurvature(AnalysisBase):
+
+    def __init__(self, universe, select='all',
+                 n_x_bins=100, n_y_bins=100,
+                 x_range=None,
+                 y_range=None,
+                 pbc=True, **kwargs):
+
+        super().__init__(universe.universe.trajectory, **kwargs)
+        self.selection = universe.select_atoms(select)
+        self.pbc = pbc
+        self.n_x_bins = n_x_bins
+        self.n_y_bins = n_y_bins
+        self.x_range = x_range if x_range else (0, universe.dimensions[0])
+        self.y_range = y_range if y_range else (0, universe.dimensions[1])
+
+        # Raise if selection doesn't exist
+        if (self.selection.n_residues) == 0:
+            raise ValueError("Invalid selection. Empty AtomGroup.")
+
+        # Raise if range doesn't cover entire dimensions of simulation box
```

Comment:
I think that's something I didn't pay too much attention to (and I didn't clarify, my bad). One of the assumptions I had is that calculation of membrane curvature is performed in equilibrated systems.  If the box shrinks or expands too much (one unit cell in the grid), the calculation wouldn't be accurate. 

The reason why I introduced the restrictions (which I changed now to a warning message) is that (1) if the surface is derived from a portion of the membrane, the calculation of curvature in the edges will be inaccurate because the calculation of a  gradient runs over adjacent values. On the other hand, (2) in the scenario in which the grid is too big, unnecessary `np.nans` will be introduced. It means that, again, the undefined adjacent values will have a negative impact on the calculation of curvature. 

I don't see a straightforward solution for NPT systems unless the calculation returns a different grid per frame and then results must be normalized over grids of the same size, and I don't have a clear idea, at least right now, of how to keep track of such variation over frames. Do you have any suggestions here?